File: Dart/test/shortestToChar_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:leetcode/shortestToChar.dart';

void main() {
  test("test shortestToChar", () {
    var input0 = ["loveleetcode", "aaab", "a"];
    var input1 = ["e", "b", "a"];
    var expected = [
      [3, 2, 1, 0, 1, 0, 0, 1, 2, 2, 1, 0],
      [3, 2, 1, 0],
      [0]
    ];

    final sln = Solution();
    for (var i = 0; i < input0.length; i++) {
      var output = sln.shortestToChar(input0[i], input1[i]);
      expect(output, expected[i]);
    }
  });
}

File: Typescript/test/shortestToChar.test.ts
import { expect, test } from '@jest/globals';
import * as solution from '../src/shortestToChar';

test('Example 1', () => {
	let input0 = "loveleetcode";
	let input1 = "e";
	let expected = [3, 2, 1, 0, 1, 0, 0, 1, 2, 2, 1, 0];
	expect(solution.shortestToChar(input0, input1)).toEqual(expected);
});

test('Example 2', () => {
	let input0 = "aaab";
	let input1 = "b";
	let expected = [3, 2, 1, 0];
	expect(solution.shortestToChar(input0, input1)).toEqual(expected);
});
